data/provider/src/main/java/tech/pegasys/teku/api/RewardCalculator.java

```diff
@@ -26,76 +22,50 @@
-import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.api.exceptions.BadRequestException;
 import tech.pegasys.teku.api.migrated.BlockRewardData;
 import tech.pegasys.teku.api.migrated.SyncCommitteeRewardData;
 import tech.pegasys.teku.bls.BLSPublicKey;
-import tech.pegasys.teku.infrastructure.ssz.SszList;
-import tech.pegasys.teku.infrastructure.ssz.collections.SszBitvector;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.SpecMilestone;
-import tech.pegasys.teku.spec.SpecVersion;
-import tech.pegasys.teku.spec.cache.IndexedAttestationCache;
-import tech.pegasys.teku.spec.config.SpecConfig;
 import tech.pegasys.teku.spec.datastructures.blocks.BeaconBlock;
 import tech.pegasys.teku.spec.datastructures.blocks.BlockContainer;
 import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.altair.SyncAggregate;
 import tech.pegasys.teku.spec.datastructures.metadata.BlockAndMetaData;
 import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
-import tech.pegasys.teku.spec.datastructures.operations.AttesterSlashing;
-import tech.pegasys.teku.spec.datastructures.operations.ProposerSlashing;
 import tech.pegasys.teku.spec.datastructures.state.SyncCommittee;
-import tech.pegasys.teku.spec.datastructures.state.Validator;
 import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
-import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair.BeaconStateAltair;
-import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair.MutableBeaconStateAltair;
 import tech.pegasys.teku.spec.datastructures.type.SszPublicKey;
-import tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor;
-import tech.pegasys.teku.spec.logic.common.statetransition.exceptions.EpochProcessingException;
-import tech.pegasys.teku.spec.logic.common.statetransition.exceptions.SlotProcessingException;
-import tech.pegasys.teku.spec.logic.versions.altair.block.BlockProcessorAltair;
+import tech.pegasys.teku.spec.logic.common.util.BlockRewardCalculatorUtil;
 
 public class RewardCalculator {
   private final Spec spec;
-  private static final Logger LOG = LogManager.getLogger();
+  private final BlockRewardCalculatorUtil blockRewardCalculatorUtil;
 
-  public RewardCalculator(Spec spec) {
+  public RewardCalculator(
+      final Spec spec, final BlockRewardCalculatorUtil blockRewardCalculatorUtil) {
     this.spec = spec;
+    this.blockRewardCalculatorUtil = blockRewardCalculatorUtil;
   }
 
-  ObjectAndMetaData<BlockRewardData> getBlockRewardData(
+  public ObjectAndMetaData<BlockRewardData> getBlockRewardDataAndMetaData(
       final BlockAndMetaData blockAndMetaData, final BeaconState parentState) {
     return blockAndMetaData.map(
-        __ -> getBlockRewardData(blockAndMetaData.getData().getMessage(), parentState));
+        __ ->
+            BlockRewardData.fromInternal(
```

Comment:
altair exception was wrapped in `BadRequestException` here too, so you need to just call local `getBlockRewardData` here

Reply:
ah ye good catch